### ISSUE TITLE
Support for HW-W

### ISSUE THREAD

**Issue by Amersel** [tangrs/nspire-linux-loader2#7]
There should be support for the new 240x360 LCD screen, since there isn't currently another way to launch Linux on those devices.

**Comment by DremOSDeveloperTeam** [comment]
This is probably my problem

**Comment by DremOSDeveloperTeam** [comment]
It took me a bit but I successfully "ported" it to the newer models (HW-W and above). It's quick and dirty so I'm not sure if it will work the best or not, but its a start.

[linuxloader2.zip](https://github.com/tangrs/nspire-linux-loader2/files/2722675/linuxloader2.zip)

**Comment by A-UNUSED-ACCOUNT** [comment]
do you have the source @DremOSDeveloperTeam

**Comment by DremOSDeveloperTeam** [comment]
I lost the source a few months ago because my hard drive died. Fortunately, you don't need the source. The source in this repo works just fine. I later realized that if you just compile LinuxLoader2 with newer versions of the Ndless SDK, it will work on models with the new screen. The issue is with the fact that there is currently no driver for the new screen in the newer models because nobody seems to really care about the TI-NSpire anymore.

**Comment by A-UNUSED-ACCOUNT** [comment]
hmm.
I still cannot get linux to boot
it hangs after the boot cmd in the .ll2
is the driver the problem

**Comment by DremOSDeveloperTeam** [comment]
Yes, that is the problem. Linux is actually booting just fine, but because of the fact that there is no display driver for the new model, you cant see anything.

**Comment by A-UNUSED-ACCOUNT** [comment]
ah, well how does one create a new driver

**Comment by DremOSDeveloperTeam** [comment]
I don't know. I don't work with low-level programming.

**Comment by A-UNUSED-ACCOUNT** [comment]
ok. this sounds like a kernel problem

**Comment by DremOSDeveloperTeam** [comment]
No. If you plug in a device into your computer and it doesnt know what it is, it is because it doesn't have a driver for it.
Device drivers are code that let the kernel interact with the hardware. Although some are built into the kernel so it has basic functionality on MOST computers (such as a typical x86 or x86_64 desktop), the TI-nSpire is not a typical computer and uses a lot of custom hardware. That's where people's own device drivers come in. These are basically add-ons that have code for the kernel to use so it, for example, can use the the custom keyboard on the TI-nSpire.
It's a driver issue. Not a kernel issue.

**Comment by A-UNUSED-ACCOUNT** [comment]
ok, is the dtb the driver file?

**Comment by DremOSDeveloperTeam** [comment]
No. The DTB contains information about the zImage, which is the actual kernel, as well as the device drivers, rootfs, applications, etc. The zImage is basically just a compressed Linux installation.

**Comment by A-UNUSED-ACCOUNT** [comment]
for now im gonna try to setup ssh over usb

**Comment by ghost** [comment]
![ApplicationFrameHost_Ynpybu0ycP](https://user-images.githubusercontent.com/18617633/108531693-f96ff880-72a4-11eb-961e-1c8e463b3a87.png)

Help me. I can't get past here ;-;

**Comment by DremOSDeveloperTeam** [comment]
No one can until someone who has a big brain can make drivers for HW-W+ and the TI-NSpire CX II.

**Comment by ghost** [comment]
sad

**Comment by kayshinonome** [comment]
nspire support is being removed from the linux kernel sadly

**Comment by ghost** [comment]
I'm not too knowledgeable about this sort of thing, but is it possible to make a rootfs that creates an SSH server and listens over USB?

**Comment by kayshinonome** [comment]
i believe so, but how to do this, is beyond me

**Comment by Xenthio** [comment]
[Archive.zip](https://github.com/tangrs/nspire-linux-loader2/files/6475489/Archive.zip)
try this. supports cx II and cx II CAS aswell

**Comment by Xenthio** [comment]
> No one can until someone who has a big brain can make drivers for HW-W+ and the TI-NSpire CX II.

read above for built binaries, source repo is here https://github.com/Vogtinator/linux/tree/cxII-clean

**Comment by ghost** [comment]
OMG thank you so much!

**Comment by kayshinonome** [comment]
Thankies ^^

**Comment by Xenthio** [comment]
enjoy my laddies

**Comment by ghost** [comment]
The screen is stil messed up but it can tell it actually works now.

**Comment by Xenthio** [comment]
screen works for me, are you using the correct dtb file?

**Comment by ghost** [comment]
I fixed it by recompiling the loader, big thanks

**Comment by ghost** [comment]
My computer even sees it over USB, wow!
![image](https://user-images.githubusercontent.com/18617633/119159165-fbbd2c80-ba24-11eb-91e0-8c5fe5ac2b8d.png)

**Comment by PirateOTC** [comment]
> 
> 
> My computer even sees it over USB, wow!
> ![image](https://user-images.githubusercontent.com/18617633/119159165-fbbd2c80-ba24-11eb-91e0-8c5fe5ac2b8d.png)

could you send me your recompiled loader?

**Comment by Vogtinator** [comment]
I updated the linuxloader2 build on https://tiplanet.org/nspire-linux-builds/ in April, it supports HW-W meanwhile.

**Comment by kayshinonome** [comment]
Is it possible to integrate these builds with github actions?

**Comment by Vogtinator** [comment]
> Is it possible to integrate these builds with github actions?

That requires some work to resurrect the Ndless toolchain container builds first (see also https://github.com/ndless-nspire/Ndless/pull/292), to use them in other repos for building.

This rebuild was (so far) the only needed change in several years, so I'd argue that continuous builds wouldn't be that useful.

**Comment by kayshinonome** [comment]
Ah I see
